src/System.Private.ServiceModel/tests/Common/Infrastructure/OS.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Infrastructure.Common
+{
+    // Helper class for OSID, an enum created to match
+    // RuntimeInformation.OSDescription
+    public static class OS
+    {
+        // These literals must match those returned from RuntimeInformation.OSDescription.
+        private static string WindowsOsName = "Microsoft Windows"; // netcore50 || win8
+        private static string WindowsPhoneOsName = "Microsoft Windows Phone";    // wpa81
+
+        // These literals must partially match what non-Windows OS's self-report
+        private static string CentOSOsName = "CentOS";
+        private static string DebianOsName = "Debian";
+        private static string FedoraOsName = "Fedora";
+        private static string OpenSUSEOsName = "OpenSUSE";
+        private static string OSXOsName = "Darwin";
+        private static string RHELOsName = "RHEL";
+        private static string UbuntuOsName = "Ubuntu";
+
+        private static OSID _currentOSID = 0;
+        private static string _currentOSDescription;
+
+        private static string CurrentOSDescription
+        {
+            get
+            {
+                if (_currentOSDescription == null)
+                {
+                    _currentOSDescription = RuntimeInformation.OSDescription;
+                }
+
+                return _currentOSDescription;
+            }
+        }
+
+        public static OSID Current
+        {
+            get
+            {
+                if (_currentOSID == 0)
+                {
+                    _currentOSID = DetectCurrentOS();
+                    Console.WriteLine(String.Format("Detected current OS as \"{0}\" from description \"{1}\"",
+                                                     _currentOSID.Name(), 
+                                                     CurrentOSDescription));
+                }
+
+                return _currentOSID;
+            }
+        }
+
+        // Extension method that returns 'true' if the given OSID
+        // includes the OSID on which this is currently executing.
+        public static bool MatchesCurrent(this OSID id)
+        {
+            return ((Current & id) != 0);
+        }
+
+        // Extension method for OSID to provide name.
+        // Format will expand to enum's name or collection if multiple.
+        public static string Name(this OSID id)
+        {
+            return Enum.Format(typeof(OSID), id, "G");
+        }
+
+        private static OSID DetectCurrentOS()
```

Comment:
Reviewers: it is my intent to replace this string matching against the OS description with a lookup table based on $(TestNugetRuntimeId) in TestProperties.  I will need to merge #1397 before I can do that.  I prefer that we merge this current PR asap to unblock @hongdai and let me switch to using TestProperties post-merge.